### PR TITLE
[LOG4J2-2688] adding support for session attributes in log4j-web

### PR DESCRIPTION
**Pull request by rmannibucau**
Follow up of https://issues.apache.org/jira/browse/LOG4J2-2523 comments, can need a new ticket though.

### CODE REVIEW

File: log4j-web/src/main/java/org/apache/logging/log4j/web/WebLookup.java [hunk 1]
```diff
@@ -91,6 +92,19 @@ public String lookup(final LogEvent event, final String key) {
...
         }
 
+        if (key.startsWith(SESSION_ATTR_PREFIX)) {
+            final ServletRequest req = getRequest();
+            if (HttpServletRequest.class.isInstance(req)) {
+                final HttpSession session = HttpServletRequest.class.cast(req).getSession(false);
+                if (session != null) {
+                    final String name = key.substring(SESSION_ATTR_PREFIX.length());
+                    final Object attribute = session.getAttribute(name);
+                    return attribute == null ? null : String.valueOf(attribute);
```
Comment (garydgregory) [inline log4j-web/src/main/java/org/apache/logging/log4j/web/WebLookup.java]: Hide the ternary: `return Objects.toString(session.getAttribute(name), null);`

**Comment by rmannibucau** [comment]
@rgoers good catch, added.

File: log4j-web/src/main/java/org/apache/logging/log4j/web/WebLookup.java [hunk 2]
```diff
@@ -91,6 +93,18 @@ public String lookup(final LogEvent event, final String key) {
                     HttpServletRequest.class.cast(req).getParameter(name) : null;
         }
 
+        if (key.startsWith(SESSION_ATTR_PREFIX)) {
+            final ServletRequest req = getRequest();
+            if (HttpServletRequest.class.isInstance(req)) {
```
Comment (jvz) [inline log4j-web/src/main/java/org/apache/logging/log4j/web/WebLookup.java]: Why the reflection? You could change this to be:

```
if (req instanceof HttpServletRequest) {
  final HttpSession session = ((HttpServletRequest) req).getSession(false);
  // ...
}
```

**Comment by rmannibucau** [comment]
Sure, created LOG4J2-2688

File: log4j-web/src/main/java/org/apache/logging/log4j/web/WebLookup.java (hunk 2 above)
Comment (rmannibucau) [reply log4j-web/src/main/java/org/apache/logging/log4j/web/WebLookup.java]: Guess it is mainly a coding habit and the preference over a debuggable method (java.lang.Class#cast). Not strong objection if you feel it must be change

**Pull request merged**